src/run/run.c

```diff
@@ -686,9 +704,13 @@ static int parse_argv(int argc, char *argv[]) {
                         break;
 
                 case ARG_BACKGROUND:
+                        if (optarg[0] != '\0' && !looks_like_ansi_color_code(optarg))
+                                return log_error_errno(-EINVAL, "Invalid background color: %s", optarg);
+
                         r = free_and_strdup_warn(&arg_background, optarg);
```

Comment:
Please introduce a helper function, say, `parse_background_argument()` in src/shared/parse-argument.[ch], then use it at all places where `--background=` option is parsed: run.c, nspawn.c, vmspawn.c, and ptyfwd-tool.c.